app/controllers/DashboardController.scala

```diff
@@ -118,10 +98,32 @@ trait DashboardController extends FrontendController with GatekeeperAuthWrapper
         approval = lastApproval(app)
         submission <- lastSubmission(app)
         admins <- administrators(app)
-        approvedApp = application(app.application, approval, admins, submission)
+        approvedApp: ApprovedApplication = application(app.application, approval, admins, submission)
       } yield Ok(approved(approvedApp))
   }
 
+  private def lastSubmission(app: ApplicationWithHistory)(implicit hc: HeaderCarrier): Future[SubmissionDetails] = {
+    val submission: StateHistory = app.history.filter(_.state == State.PENDING_GATEKEEPER_APPROVAL)
+      .sortWith(StateHistory.ascendingDateForAppId)
+      .lastOption.getOrElse(throw new InconsistentDataState("pending gatekeeper approval state history item not found"))
+
+    developerConnector.fetchByEmail(submission.actor.id).map(s =>
+      SubmissionDetails(s"${s.firstName} ${s.lastName}", s.email, submission.changedAt))
+  }
+
+  private def applicationDetails(app: ApplicationResponse, submission: SubmissionDetails) = {
+    ApplicationDetails(app.id.toString, app.name, app.description.getOrElse(""), submission)
+  }
+
+  def developersPage: Action[AnyContent] = requiresRole(Role.APIGatekeeper) {
+    implicit request => implicit hc =>
+      for {
+        devs <- developerConnector.fetchAll
+        (h,t) = (devs.head, devs.tail)
+        emails = t.foldLeft(h.email)((s, u) => s"$s,${u.email}")
+      } yield Ok(developers(devs, emails))
```

Comment:
Reading this, it looks like we're trying to get a comma separated list of emails? 

Would it make sense to use `devs.mkString(",")` for this?  It won't throw if the list is empty, and is shorter/clearer to boot